src/Command/DrupalUpdate/FileSystemUtility.php

```diff
@@ -0,0 +1,224 @@
+    }
+    if ($package == 'drupal') {
+      $this->io->note("Starting Drupal core update");
+      $this->updateDrupalCore($save_to . '/drupal');
+      $this->fileSystem->remove($save_to);
+    }
+  }
+
+  /**
+   * Update drupal core files with latest updates.
+   * Drupal core modules, themes and profiles.
+   * @param $core_dir_path
+   */
+  protected function updateDrupalCore($core_dir_path) {
+    $ignore_files = [
+        '.gitignore',
+        '.htaccess',
+        'CHANGELOG.txt',
+        'sites',
+    ];
+    $replace_dir_path = str_replace('/temp_drupal_core/drupal', '', $core_dir_path);
+    $finder = new Finder();
+    $finder->in($core_dir_path)->ignoreVCSIgnored(TRUE)->notPath($ignore_files)->depth('== 0')->sortByName();
+    foreach ($finder as $file) {
+      $file_name_with_extension = $file->getRelativePathname();
+      $temp_core_path = $core_dir_path . "/" . $file_name_with_extension;
+      if (is_dir($temp_core_path)) {
+        $this->fileSystem->mirror($temp_core_path, $replace_dir_path . '/' . $file_name_with_extension);
+        continue;
+      }
+      $this->fileSystem->copy($temp_core_path, $replace_dir_path . '/' . $file_name_with_extension, TRUE);
+    }
+  }
+
+  /**
+   * Remove downloaded tar.gz files.
+   * @param $remove_file_list
+   */
+  public function unlinkTarFiles($remove_file_list) {
+
+    foreach ($remove_file_list as $key => $value) {
+      if ( ($key == 0) || ($value['package_type'] == 'core') ) {
+        continue;
+      }
+      if (is_array($value['file_path'])) {
+        foreach ($value['file_path'] as $item) {
+          $this->removeFile($item . "/" . $value['package'] . ".tar.gz");
+        }
+      }
+      else {
+        $this->removeFile($value['file_path'] . "/" . $value['package'] . ".tar.gz");
+      }
+    }
+  }
+
+  /**
+   * @param $file_path
+   */
+  protected function removeFile($file_path) {
+    if ($this->fileSystem->exists($file_path)) {
+      $this->fileSystem->remove($file_path);
+    }
+    else {
+      $this->io->note("File " . $file_path . " does not exist.");
+    }
+  }
+
+  /**
+   * @param $file_url
+   * @param $save_to
+   * @param $package
+   * @return false|string|string[]
+   * @throws AcquiaCliException
+   */
+  protected function dumpPackageTarFile($file_url, $save_to, $package) {
+    try {
+      if ($this->downloadFileGuzzleClient($file_url, $save_to . '/' . $package . '.tar.gz')) {
+        return str_replace('.tar.gz', '', basename($file_url));
+      }
+    }
+    catch (Exception $e) {
+      // @todo handle errors
+      throw new AcquiaCliException("Unable to download {$package} file.");
+    }
+    return FALSE;
+  }
+
+  /**
+   * @param $file_url
+   * @param string $method
+   * @param string $header_type
+   * @return false|mixed
+   * @throws AcquiaCliException
+   * @throws GuzzleException
+   */
+  public function getFileContentsGuzzleClient($file_url, $method = 'GET', $header_type = '') {
```

Comment:
Missing both function's return type declaration and parameters' type declaration